src/org/python/core/PyByteArray.java

```diff
@@ -1335,17 +1326,14 @@ public PyObject __iadd__(PyObject o) {
 
     @ExposedMethod(type = MethodType.BINARY, doc = BuiltinDocs.bytearray___iadd___doc)
     final synchronized PyObject bytearray___iadd__(PyObject o) {
-        PyType oType = o.getType();
-        if (oType == TYPE) {
-            // Use the general method, specifying the crack at the end of the array.
-            // Note this deals with the case o==this.
-            setslice(size, size, 1, (BaseBytes)o);
```

Comment:
As the comment points out, as well as being an easy optimisation, this deals with the case `a += a`, which I think this can't. Although ... CPython makes the same error. (Well, **I** think it's an error.)